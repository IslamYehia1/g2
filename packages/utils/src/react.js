import React from 'react';
import { is } from './is';

export function getValidChildren(children) {
	return React.Children.toArray(children).filter((child) =>
		React.isValidElement(child),
	);
}

export function assignRef(ref, value) {
	if (ref == null) return;

	if (is.function(ref)) {
		ref(value);
		return;
	}

	try {
		//@ts-ignore
		ref.current = value;
	} catch (error) {
		throw new Error(`Cannot assign value '${value}' to ref '${ref}'`);
	}
}

export function mergeRefs(...refs) {
	return (value) => {
		refs.forEach((ref) => assignRef(ref, value));
	};
}
