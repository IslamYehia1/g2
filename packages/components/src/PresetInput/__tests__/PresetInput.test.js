import { render } from '@testing-library/react';
import React from 'react';

import { PresetInput } from '../index';

describe('props', () => {
	test('should render correctly', () => {
		const { container } = render(<PresetInput />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
