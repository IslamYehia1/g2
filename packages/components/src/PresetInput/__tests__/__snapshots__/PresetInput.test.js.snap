// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
<div
  class="css-1f55os3"
>
  <div
    class="wp-components-flex wp-components-base-field wp-components-text-input wp-components-unit-input wp-components-preset-input css-in42v"
    data-g2-c16t="true"
    data-g2-component="TextInputWrapper"
  >
    <input
      autocomplete="off"
      class="css-16e6wng"
      data-g2-c16t="true"
      data-g2-component="TextInput"
      spellcheck="false"
      type="text"
      value=""
    />
    <div
      class="wp-components-flex-item css-19kxsx5"
      data-g2-c16t="true"
      data-g2-component="TextInputSuffix"
    >
      <div
        class="css-1ilaesb"
      >
        <div
          class="css-u6dr0v"
        >
          <span
            class="wp-components-truncate wp-components-text css-12kpubv"
            data-g2-c16t="true"
            data-g2-component="Text"
          >
            <div
              class="wp-components-icon css-1d07aa5"
              data-g2-c16t="true"
              data-g2-component="Icon"
            >
              <svg
                fill="none"
                height="12"
                size="12"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="6 9 12 15 18 9"
                />
              </svg>
            </div>
          </span>
        </div>
        <select
          class="css-1ienp1t"
        />
      </div>
    </div>
  </div>
</div>
`;
